autocorpus/parse_xml.py

```diff
@@ -648,7 +647,7 @@ def find_parent_titles(title_element: Tag | NavigableString) -> list[str]:
                 matches = re.findall(pattern, text_test)
 
                 # Remove duplicate closing tags and create a list of unique matches
-                good_matches = list(np.unique(matches))
+                good_matches = list(dict.fromkeys(matches))
```

Comment:
Would a `set` have made more sense here? I'm a bit thrown by making a dictionary and immediately casting it to a list

Reply:
That was my first thought, but the problem is that `set`s are unordered so your matches will end up jumbled up. Might not be a problem here, but I didn't want to change the behaviour.

Credit/blame for the `dict` hack goes to SO: https://stackoverflow.com/questions/12897374/get-unique-values-from-a-list-in-python